docs/topics/classes.md

```diff
@@ -1,263 +1,587 @@
+isn't enough.
 
-```kotlin
-class Person(val pets: MutableList<Pet> = mutableListOf())
+Secondary constructors are useful for [Java interoperability](java-to-kotlin-interop.md) and when
+you need multiple ways to initialize a class.
+
+To declare a secondary constructor, use the `constructor`
+keyword inside the class body with the constructor parameters within parentheses. Use curly braces with the constructor logic inside:
 
-class Pet {
-    constructor(owner: Person) {
-        owner.pets.add(this) // adds this pet to the list of its owner's pets
+```kotlin
+// Class header with a primary constructor that initializes name and age
+class Person(val name: String, var age: Int) {
+
+    // Secondary constructor that takes a String age
+    // and converts it to an integer
+    constructor(name: String, age: String) : this(name, age.toIntOrNull() ?: 0) {
+        println("$name created with converted age: $age")
+        // Bob created with converted age: 8
     }
 }
+
+fun main() {
+    // Uses the secondary constructor with a String age
+    Person("Bob", "8")
+}
 ```
+{kotlin-runnable="true"}
+
+> The expression `age.toIntOrNull() ?: 0` converts the `age` string to an integer. For more information about this expression, see [Null safety](null-safety.md).
+>
+{style="tip"}
+
+In the code above, the secondary constructor delegates to the primary constructor via the `this` keyword, 
+passing `name` and the `age` value converted to an integer.
+
+In Kotlin, secondary constructors must delegate to the primary constructor. This delegation ensures that all primary 
+constructor initialization logic is executed before any secondary constructor logic runs. 
+
+Constructor delegation can be: 
+* **Direct**, where the secondary constructor calls the primary constructor immediately.
+* **Indirect**, where one secondary constructor calls another, which in turn delegates to the primary constructor.
 
-If the class has a primary constructor, each secondary constructor needs to delegate to the primary constructor, either
-directly or indirectly through another secondary constructor(s). Delegation to another constructor of the same class is
-done using the `this` keyword:
+Here's an example demonstrating how direct and indirect delegation works:
 
 ```kotlin
-class Person(val name: String) {
-    val children: MutableList<Person> = mutableListOf()
-    constructor(name: String, parent: Person) : this(name) {
-        parent.children.add(this)
+// Class header with a primary constructor that initializes name and age
+class Person(
+    val name: String,
+    var age: Int
+) {
+    // Secondary constructor with direct delegation 
+    // to the primary constructor
+    constructor(name: String) : this(name, 0) {
+        println("Person created with default age: $age and name: $name.")
+        // Person created with default age: 0 and name: Alice.
```

Comment:
Can you move the code comments to where they will appear when triggered by the main function?